src/select-menu/src/OptionsList.js

```diff
@@ -164,6 +165,10 @@ export default class OptionsList extends PureComponent {
     if (e.keyCode === 13) {
       this.handleEnter()
     }
+
+    if (e.keyCode === 27) {
```

Comment:
Instead of `keyCode` we should use `key`: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

```suggestion
    if (e.key === 'Escape') {
```